app/controllers/SchemeVarianceLockCacheController.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import com.google.inject.Inject
+import models.SchemeVariance
+import play.api.Configuration
+import play.api.libs.json.Json
+import play.api.mvc._
+import repositories.LockRepository
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.http.BadRequestException
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SchemeVarianceLockCacheController @Inject()(
+                                                   config: Configuration,
+                                                   repository: LockRepository,
+                                                   val authConnector: AuthConnector,
+                                                   cc: ControllerComponents
+                                                 )(implicit ec: ExecutionContext) extends BackendController(cc) with AuthorisedFunctions {
+
+  def lock(): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised() {
+        getIds { (psaId, srn) =>
```

Comment:
Better name would be "withIDs", i.e. with ids do x...

Reply:
fixed